### PR TITLE
Add an Intent Action handling for when Neeva is the Default Search Provider

### PR DESCRIPTION
**Pull request by Kokobe**
Adds an action for the OS to use when the Search Provider is Neeva. 

The `action` we are adding and should be submitting is: `com.neeva.app.PERFORM_SEARCH`

### Pointers to help with review
* clickup + context documents: https://app.clickup.com/t/36034046/RST-1535

### How was this tested?
- [x] I tested this manually on `debug`
- [ ] I tested this manually on `staging` or `release`
- [x] I added automated tests
- [ ] Existing automated tests cover some of my changes
- [ ] Existing automated tests cover all of my changes
- [x] I checked that `@Previews` still work in Android Studio

### Manual test cases
Testing using the terminal:

https://user-images.githubusercontent.com/20916043/221061173-0437d770-a5c0-408f-bf34-2d8e0dc7b7b9.mov

### CODE REVIEW

File: app/src/androidTest/java/com/neeva/app/searchprovider/DefaultSearchProviderTest.kt [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    @get:Rule(order = 10000)
+    val androidComposeRule = AndroidComposeTestRule(
+        activityRule = TestRule { base, _ -> base },
+        activityProvider = { multiActivityTestRule.getLastForegroundActivity()!! }
+    )
+
+    val neevaActivityComposeRule = androidComposeRule
+        as AndroidComposeTestRule<ActivityScenarioRule<NeevaActivity>, NeevaActivity>
+
+    private fun assertThatNeevaActivityIsOpen() {
```
Comment (dan-alcantara-neeva) [inline app/src/androidTest/java/com/neeva/app/searchprovider/DefaultSearchProviderTest.kt:61]: Is this a fork/copy from some other file?

**Review by dan-alcantara-neeva — COMMENTED** [review]
Have you tested what happens whenyou're on other screens and the app is alive, when the app is dead, when first run hasn't been done yet...

File: app/src/androidTest/java/com/neeva/app/searchprovider/DefaultSearchProviderTest.kt (hunk 1 above)
Comment (Kokobe) [reply app/src/androidTest/java/com/neeva/app/searchprovider/DefaultSearchProviderTest.kt:61]: Yeah. I'll make it a function that both files can use

**Comment by Kokobe** [comment]
I did test when FirstRun hasn't been done yet (see `firstRunNotDone` tests) but I'll ensure that they can finish the WelcomeFlowActivity properly.

How would I instrumentation test what would happen if I started the intent when the app is dead? 

I'll add a test for when I'm on another screen 

@dan-alcantara-neeva

**Comment by Kokobe** [comment]
Ended up manually testing what happens when you get that intent in the middle of FirstRun or if you were on a different screen in the app, it works as intended.

**Review by dan-alcantara-neeva — APPROVED** [review]

**Pull request merged**